### PR TITLE
Plot update

### PR DESCRIPTION
**Pull request by alexandersundin**

### CODE REVIEW

File: source/plot/Axis.ooc [hunk 1]
```diff
@@ -42,84 +42,114 @@ Axis: class {
 		result
 	}
 
-	getSVG: func (plotAreaSize, axisAreaSize, position, translationToOrigo, scaling: FloatPoint2D) -> String {
+	getSvg: func (plotAreaSize, margin, translationToOrigo, scaling: FloatPoint2D, fontSize: Int) -> String {
```
Comment (davidhesselbom) [inline source/plot/Axis.ooc]: Pretty sure these variables are `FloatSize2D`, not `FloatPoint2D`.

File: source/plot/Axis.ooc [hunk 2]
```diff
@@ -42,84 +42,114 @@ Axis: class {
...
+	}
 
-			tickValue := this getFirstTickValue()
-			position y += translationToOrigo y - plotAreaSize y - scaling y * tickValue
-			while (tickValue <= this max) {
-				result = result & this getTickSVG(tickValue, radix, position, numberOffset, rightTickMarkerStartOffset, tickMarkerEndOffset, "end")
-				tickValue += this tick
-				position y += - scaling y * tick
-			}
+	getHorizontalSvg: func(plotAreaSize, margin, translationToOrigo, scaling, position: FloatPoint2D, radix: Float) -> String {
```
Comment (davidhesselbom) [inline source/plot/Axis.ooc]: Same here...

File: source/plot/Axis.ooc [hunk 3]
```diff
@@ -42,84 +42,114 @@ Axis: class {
...
 		result
 	}
 
-	getFirstTickValue: func -> Float {
-		result: Float
-		if ((this min < this tick && this min > 0.0f) || (this min > this tick && this min > 0.0f && this min < 1.0f))
-			result = this min + (this tick - Float modulo(this min, this tick))
-		else
-			result = this min - Float modulo(this min, this tick)
+	getVerticalSvg: func(plotAreaSize, margin, translationToOrigo, scaling, position: FloatPoint2D, radix: Float) -> String {
```
Comment (davidhesselbom) [inline source/plot/Axis.ooc]: And here.

File: source/plot/Axis.ooc [hunk 4]
```diff
@@ -42,84 +42,114 @@ Axis: class {
...
+		result = result & Shapes text(position + numberOffset, Float absolute(tickValue - tickValue round()) < 0.001 ? tempTickInt : tempTick, this fontSize - 4, textAnchor)
 		tempTick free()
 		tempTickInt free()
 		result = result >> "</g>\n"
 		result
 	}
 
+	getFirstTickValue: func -> Float {
+		result: Float
+		if ((this min < this tick && this min > 0.0f) || (this min > this tick && this min > 0.0f && this min < 1.0f))
```
Comment (davidhesselbom) [inline source/plot/Axis.ooc]: Can `tick` be < 0?

File: source/plot/ScatterPlotData2D.ooc [hunk 5]
```diff
@@ -29,14 +29,14 @@ ScatterPlotData2D: class extends PlotData2D {
 		super(xSeries, ySeries, label, colorBgra)
 	}
 
-	getSVG: func (scaling: FloatPoint2D) -> String {
+	getSvg: func (scaling: FloatPoint2D) -> String {
```
Comment (davidhesselbom) [inline source/plot/ScatterPlotData2D.ooc]: And here. `scaling` should be a `FloatSize2D`.

File: source/plot/SvgPlot.ooc [hunk 6]
```diff
@@ -127,15 +131,15 @@ SVGPlot: class {
 	}
 
 	setLegends: func (size, plotAreaSize: FloatPoint2D) -> String {
-		result := "<svg desc='Legends' x='" clone() & ((size x - plotAreaSize x) / 2) toString() & "' y='" clone() & ((size y - plotAreaSize y) / 2) toString() & "' width='" clone() & plotAreaSize x toString() & "' height='" clone() & plotAreaSize y toString() & "' xmlns:xlink='http://www.w3.org/1999/xlink' xmlns:drag='http://www.codedread.com/dragsvg' onload='initializeDraggableElements();' onmouseup='mouseUp(evt)' onmousemove='mouseMove(evt)'>\n<script id='draggableLibrary' xlink:href='http://www.codedread.com/dragsvg.js'/>\n<g id='Legend' drag:enable='true'>\n" clone()
+		result := "<svg desc='Legends' x='" << ((size x - plotAreaSize x) / 2) toString() >> "' y='" & ((size y - plotAreaSize y) / 2) toString() >> "' width='" & plotAreaSize x toString() >> "' height='" & plotAreaSize y toString() >> "' xmlns:xlink='http://www.w3.org/1999/xlink' xmlns:drag='http://www.codedread.com/dragsvg' onload='initializeDraggableElements();' onmouseup='mouseUp(evt)' onmousemove='mouseMove(evt)'>\n<script id='draggableLibrary' xlink:href='http://www.codedread.com/dragsvg.js'/>\n<g id='Legend' drag:enable='true'>\n"
```
Comment (davidhesselbom) [inline source/plot/SvgPlot.ooc]: And here.

**Comment by alexandersundin** [comment]
Updated

**Comment by alexandersundin** [comment]
Updated

File: source/math/FloatExtension.ooc [hunk 7]
```diff
@@ -126,17 +126,17 @@ extend Float {
...
+	getScientificPowerString: static func (value: This) -> String {
 		(coefficient, radix) := This decomposeToCoefficientAndRadix(value, 1)
 		power := log10(radix) as Int
 		result := ""
-		if (coefficient != 1.0f)
-			result = result & coefficient toString() & "×" clone()
-		result = result & "10^" clone() & power toString()
+		if (This absolute(coefficient - 1.0f) > This epsilon)
+			result = result & coefficient toString() >> "×"
+		result = result >> "10^" & power toString()
```
Comment (davidhesselbom) [inline source/math/FloatExtension.ooc]: Wouldn't it be prettier to write `E` instead of `10^`?

Comment (davidhesselbom) [inline source/math/FloatExtension.ooc]: And if you want to keep `10`, there should really be a space surrounding the `×`, and the `^exponent` should be replaced with `<sup>exponent</sup`.

File: source/math/FloatExtension.ooc [hunk 8]
```diff
@@ -126,17 +126,17 @@ extend Float {
...
 	}
-	getScientificPowerString: static func (value: Float) -> String {
+	getScientificPowerString: static func (value: This) -> String {
 		(coefficient, radix) := This decomposeToCoefficientAndRadix(value, 1)
 		power := log10(radix) as Int
 		result := ""
-		if (coefficient != 1.0f)
-			result = result & coefficient toString() & "×" clone()
-		result = result & "10^" clone() & power toString()
+		if (This absolute(coefficient - 1.0f) > This epsilon)
```
Comment (davidhesselbom) [inline source/math/FloatExtension.ooc]: I still think this is unnecessary. You should only need to check that `This absolute(coefficient) != 1.0f`. If the difference between `coefficient` and `1.0f` is smaller than `This epsilon`, then `coefficient` **is** equal to 1.0f.

**Comment by alexandersundin** [comment]
Updated

**Comment by alexandersundin** [comment]
Updated

**Pull request merged**